transformation_test/transformation_test.go

```diff
@@ -90,50 +89,48 @@ func TestTransformationTests(t *testing.T) {
 				fmt.Sprintf("--config=%s", filepath.Join(tempPath, flbMainConf)),
 				fmt.Sprintf("--parser=%s", filepath.Join(filepath.Join(tempPath, flbParserConf))))
 
-			var stdout, stderr io.ReadCloser
-			stdout, err = cmd.StdoutPipe()
-			if err != nil {
-				t.Fatal("stdout pipe failure", err)
-			}
-			stderr, err = cmd.StderrPipe()
-			if err != nil {
-				t.Fatal("stderr pipe failure", err)
-			}
-
-			if err := cmd.Start(); err != nil {
-				t.Fatal("Failed to start command:", err)
+			var stdout, stderr bytes.Buffer
+			cmd.Stdout = &stdout
+			cmd.Stderr = &stderr
+			if err := cmd.Run(); err != nil {
+				t.Log(stderr.String())
+				t.Fatal("Failed to run command:", err)
 			}
 
-			// stderr and stdout need to be read in parallel to prevent deadlock
-			var eg errgroup.Group
-			eg.Go(func() error {
-				// read stderr
-				slurp, _ := io.ReadAll(stderr)
-				t.Logf("stderr: %s\n", slurp)
-				return nil
-			})
-
-			// read and unmarshal output
-			var data []map[string]any
-			out, _ := io.ReadAll(stdout)
-			_ = eg.Wait()
+			var d []map[string]any
+			var data []map[string]interface{}
 
-			err := yaml.Unmarshal(out, &data)
-			if err != nil {
-				t.Log(string(out))
-				t.Fatal(err)
+			sc := bufio.NewScanner(bytes.NewReader(stdout.Bytes()))
```

Comment:
You should use a `json.Decoder` to read the sequential JSON maps, instead of using a `bufio.Scanner` + `yaml.Unmarshal`